### PR TITLE
PR: Add file hover-go-to relative to project root

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

- [x] Fixes file:// handling in the editor with "~/paths..."
- [x] Add support for project root path using `^/` to denote project root path
* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by goanpeca** [comment]
Hi, @nerohmot. This PR adds now support for `^/` project root path. Also fixes handling of `~/ `  which was not working. I will check the `#page` on pdf files in a separate PR.

File: spyder/plugins/editor/widgets/codeeditor.py [hunk 1]
```diff
@@ -4191,11 +4205,25 @@ def _preprocess_file_uri(self, uri):
         fname = uri.replace('file://', '')
         if fname[-1] == '/':
             fname = fname[:-1]
+
+        # ^/ is used to denote current project root
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/codeeditor.py:4209]: ... to denote **the** current ...

File: spyder/plugins/editor/widgets/editor.py [hunk 2]
```diff
@@ -1396,6 +1396,19 @@ def set_run_cell_copy(self, state):
...
+        Set the current active project root path.
+
+        Parameters
+        ----------
+        root_path: str or None, optional
+            Path to current project root path. Default is None.
+        """
+        for index in range(self.get_stack_count()):
+            editor = self.tabs.widget(index)
+            editor.set_current_project_path(root_path)
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/editor.py:1410]: Why not using here `for finfo in self.data:`? That's what we use to iterate among CodeEditors in almost all methods in `EditorStack`.

Comment (goanpeca) [reply spyder/plugins/editor/widgets/editor.py:1410]: Fixed

File: spyder/plugins/editor/widgets/codeeditor.py (hunk 1 above)
Comment (goanpeca) [reply spyder/plugins/editor/widgets/codeeditor.py]: Fixed

**Pull request merged**